providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProvider.java

```diff
@@ -0,0 +1,263 @@
+    public List<Hook> getProviderHooks() {
+        return FeatureProvider.super.getProviderHooks();
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(
+        String key, Boolean defaultValue, EvaluationContext evaluationContext) {
+        // When isUsingBooleanConfigValue the feature_state_value will be used as the flag value
+        if (this.options.isUsingBooleanConfigValue()) {
+            return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Boolean.class);
+        }
+        // Otherwise the enabled field will be used as the boolean flag value
+        try {
+            Flags flags = getFlags(evaluationContext);
+            boolean isFlagEnabled = flags.isFeatureEnabled(key);
+            Reason reason = isFlagEnabled ? null : Reason.DISABLED;
+            return buildEvaluation(isFlagEnabled, null, reason, null);
+        } catch (FlagsmithClientError flagsmithClientError) {
+            return buildEvaluation(defaultValue, ErrorCode.GENERAL, Reason.ERROR, null);
+        }
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(
+        String key, String defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, String.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(
+        String key, Integer defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Integer.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(
+        String key, Double defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Double.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(
+        String key, Value defaultValue, EvaluationContext evaluationContext) {
+        return resolveFlagsmithEvaluation(key, defaultValue, evaluationContext, Value.class);
+    }
+
+    /**
+     * Get all the flags for a given environment or identity. The Method will use
+     * getEnvironmentFlags from the Flagsmith sdk if no targeting key is provided
+     * in the EvaluationContext. If a targeting key is provided then the
+     * getIdentityFlags method will be used.
+     *
+     * @param ctx an EvaluationContext object with flag evaluation options
+     * @return a Flagsmith Flags object with all the respective flags
+     * @throws FlagsmithClientError Thrown when there are issue retrieving the flags
+     *                              from Flagsmith
+     */
+    private Flags getFlags(EvaluationContext ctx) throws FlagsmithClientError {
+        return Objects.isNull(ctx.getTargetingKey()) || ctx.getTargetingKey().isEmpty()
+            ? flagsmith.getEnvironmentFlags()
+            // Todo add traits when attributes are added to context
```

Comment:
I think I would recommend we implement traits before we release this, or context functionality won't work as expected. Let me know if I've misunderstood.

Reply:
We've managed to implement this now